src/Uno.UI/DataBinding/BindingPath.cs

```diff
@@ -478,15 +478,41 @@ internal object GetSubstituteValue()
 				return GetSourceValue(_substituteValueGetter);
 			}
 
+			private bool _isDataContextChanging;
+
 			private void OnDataContextChanged()
 			{
-				if (DataContext != null)
+				if (DataContext != null && DataContext != DependencyProperty.UnsetValue)
 				{
 					ClearCachedGetters();
+					if (_propertyChanged.Disposable != null)
+					{
+#if !HAS_EXPENSIVE_TRYFINALLY // Try/finally incurs a very large performance hit in mono-wasm - https://github.com/mono/mono/issues/13653
+						try
+#endif
+						{
+							_isDataContextChanging = true;
+							_propertyChanged.Disposable = null;
+						}
+#if !HAS_EXPENSIVE_TRYFINALLY // Try/finally incurs a very large performance hit in mono-wasm - https://github.com/mono/mono/issues/13653
+						finally
+#endif
+						{
+							_isDataContextChanging = false;
+						}
+					}
 
 					_propertyChanged.Disposable =
 							SubscribeToPropertyChanged((previousValue, newValue, shouldRaiseValueChanged) =>
 								{
+									if (_isDataContextChanging && newValue == DependencyProperty.UnsetValue)
```

Comment:
I'm wondering if testing for the type would be faster than calling `object.Equals` there. Not probably a big deal either.

Reply:
Now that we have a type for that... but it would be a pattern that cannot be replicated outside the Uno's code. Maybe we should stick with code that can be used elsewhere...

Reply:
Done!